### ISSUE TITLE
"apply model" fuctionality

### ISSUE THREAD

**Issue by kliegr** [lvhimabindu/interpretable_decision_sets#1]
I did not find in the current code base a function for applying the learnt rule model. 

If it is indeed not available I am ready to  implement the one described in:
Lakkaraju, Himabindu, Stephen H. Bach, and Jure Leskovec. "Interpretable decision sets: A joint framework for description and prediction." Proceedings of the 22nd ACM SIGKDD International Conference on Knowledge Discovery and Data Mining. ACM, 2016

Do I understand  correctly that for the experiments performed in the paper (as described in last para of sec 3.1), the following "apply model" procedure was followed?
1) for each rule, compute the F1 measure over training data
2) sort rules according to the F1 measure
3) for a given test instance, find the first rule with matching antecedent in the list and assign its consequent as a label to the instance.
4) if no such rule is found, apply majority class in the training data.

Last question, is the F1 measure computed as a harmonic mean of rule confidence (precision) and rule support (recall)?

Thanks!

**Comment by ntddk** [comment]
Hi,

I got the same problem while replicating the KDD'16 paper.
Or have you already (re)-implemented its functionality? @kliegr

With Best Regards

**Comment by kliegr** [comment]
No, 
I have not yet reimplemented this.
I you do @ntddk, please let me know.

Tomas
